### PR TITLE
CR-1098839: Fixing profiling regression on non-AIE Versal designs

### PR DESCRIPTION
**Pull request by jvillarre**
A recent checkin added information specific to AIE designs into the profiling code in the xdp_core library.  On applications where the AIE library is not loaded, this caused the programs to fail when loading xdp_core due to unresolved references.

This pull request moves the AIE specific implementation out of xdp_core and into the aie plugins, while still storing the information in the xdp_core database.  In order to accomplish this, the database now needs additional function pointers to know how to allocate and deallocate the specific pointers as they are stored as void*.

### CODE REVIEW

**Review by pgschuey — APPROVED** [review]
Looks good

File: src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp [hunk 1]
```diff
@@ -34,6 +34,36 @@
...
+    if (!aieArray)
+      return nullptr ;
+    return aieArray->getDevInst() ;
+  }
+
+  static void* allocateAieDevice(void* devHandle)
+  {
+    XAie_DevInst* aieDevInst =
+      static_cast<XAie_DevInst*>(fetchAieDevInst(devHandle)) ;
+    if (aieDevInst != nullptr)
```
[the remark this answers is not in the document]
Comment (jvillarre) [reply src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp]: Changed to (!aieDevInst) in latest commit

File: src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_plugin.cpp [hunk 2]
```diff
@@ -38,6 +38,35 @@
...
+    if (!aieArray)
+      return nullptr ;
+    return aieArray->getDevInst() ;
+  }
+
+  static void* allocateAieDevice(void* devHandle)
+  {
+    XAie_DevInst* aieDevInst =
+      static_cast<XAie_DevInst*>(fetchAieDevInst(devHandle)) ;
+    if (aieDevInst != nullptr)
```
[the remark this answers is not in the document]
Comment (jvillarre) [reply src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_plugin.cpp]: Changed to (!aieDevInst) in latest commit

**Review by IshitaGhosh — APPROVED** [review]
The changes look good.

**Review by hackwa — APPROVED** [review]
looks good

**Pull request merged**